docs/upgrades.md

```diff
@@ -0,0 +1,196 @@
+# Upgrades
+
+This page contains some guidance on upgrading an existing deployment of Sematic
+in various circumstances.
+
+## General guidance
+
+Sematic has 3 components, depending on your usage:
+
+- *pip package*: If you are using Sematic only locally, this is
+the only component you need.
+- *Server*: If you are using Sematic in the cloud, you need this
+and the pip package.
+- *bazel integration*: If you are using `bazel` with Sematic, you
+need this component as well.
+
+When you are upgrading Sematic, you generally want to upgrade all of
+the components you are using at the same time. It is impossible to
+upgrade all of them literally atomically, so we reccommend upgrading in
+the order:
+
+- Server
+- bazel integration
+- pip package
+
+Newer servers are often compatible with older pip packages, but pip
+packages cannot be newer than the server they are interacting with.
+
+To see which pip packages a given server version supports, check the
+[release notes](https://github.com/sematic-ai/sematic/releases).
+
+Pip packages and the bazel integration should always be kept in sync.
+
+### Upgrading the pip package
+
+If you have installed via command line, you can do the following
+to upgrade to the latest:
+
+```bash
+pip install sematic --upgrade
+```
+
+or to install a specific version:
+
+```bash
+pip install sematic==X.Y.Z
+```
+
+If you use a `requirements.txt` file or similar, you can increase the version
+there and then install from the `requirements.txt` file.
+
+### Upgrading the bazel integration
+
+Your bazel `WORKSPACE` should contain something like the following:
+
+```python
+git_repository(
+    name = "rules_sematic",
+    remote = "https://github.com/sematic-ai/sematic.git",
+    strip_prefix = "bazel",
+    tag = "vX.Y.Z",
+)
+```
+
+To upgrade, change the `tag` to the version you wish to install (should match
+the pip package).
+
+### Upgrading the Server
+
+The server should be installed via `helm`. Make any adjustments to your `values.yaml`
+required by the latest version (See the section
+["From vX1.X2.X3 to vY1.Y2.Y3"](#from-vx1x2x3-to-vy1y2y3) below). Then:
+
+```bash
+helm upgrade sematic-server -f /path/to.values.yaml
+```
+
+## Upgrading the server past supported client versions
+
+You will always want to upgrade the server before upgrading the clients. If the
```

Comment:
Please define what the clients are. Maybe they should be listed in the list under "General guidance"